pkg/ebpf/tracer_test.go

```diff
@@ -238,6 +240,91 @@ func TestPreexistingConnectionDirection(t *testing.T) {
 	doneChan <- struct{}{}
 }
 
+func TestIncomingTCPConnectionDirection(t *testing.T) {
+	tr, err := NewTracer(NewDefaultConfig())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tr.Stop()
+
+	tr.portMapping.AddMapping(8000)
+	connStats := ConnectionStats{}
+	connStats.Source = util.AddressFromString("10.0.2.25")
+	connStats.SPort = 8000
+	connStats.Dest = util.AddressFromString("38.122.226.210")
+	connStats.DPort = 5893
+	connStats.Type = TCP
+
+	conn := []ConnectionStats{connStats}
+	tr.setConnectionDirections(conn)
+	assert.Equal(t, INCOMING, conn[0].Direction)
+}
+
+func TestOutgoingTCPConnectionDirection(t *testing.T) {
+	tr, err := NewTracer(NewDefaultConfig())
```

Comment:
nit: instead of using `NewTracer()`  which will attach a BPF program, you can do something like:

```go
tr := &tracer{
   // initialize anything necessary here
}
```

And move this to a test file that isn't tagged as `linux_bpf`. That way these tests will run as part of a normal `go test`

Reply:
Moved other tests to a file and do note enable a full tracer. This test remains as an integration test.